spec/unit/git/rugged/credentials_spec.rb

```diff
@@ -79,6 +79,36 @@
     end
   end
 
+  describe "generating github app tokens" do
+    it 'errors if app id has invalid characters' do
+      R10K::Git.settings[:github_app_id] = "123A567890"
```

Comment:
Are all these `R10K::Git.settings` necessary if we're just passing params directly to the `github_app_token` method? I think this test is passing because you're using `nil`, not because of this setting. You could also just pass `R10K::Git.settings[:github_app_id]` to the method, like you do in the 'private file does not exist' test.